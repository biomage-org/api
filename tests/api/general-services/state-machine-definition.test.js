const { buildStateMachineDefinition } = require('../../../src/api/general-services/pipeline-manage');
const { getQcPipelineSkeleton } = require('../../../src/api/general-services/pipeline-manage/skeletons/qc-pipeline-skeleton');
const { getGem2sPipelineSkeleton } = require('../../../src/api/general-services/pipeline-manage/skeletons/gem2s-pipeline-skeleton');

jest.mock('crypto', () => ({
  ...jest.requireActual('crypto'),
  randomBytes: () => Buffer.from('asdfg'),
}));

const snapshotPlainJsonSerializer = {
  // eslint-disable-next-line no-unused-vars
  test(val) {
    return true;
  },
  // eslint-disable-next-line no-unused-vars
  serialize(val, prettyConfig, indentation, depth, refs, printer) {
    return `\n${JSON.stringify(val, null, 2)}`;
  },
};
expect.addSnapshotSerializer(snapshotPlainJsonSerializer);


describe('non-tests to document the State Machines', () => {
  const context = {
    experimentId: 'mock-experiment-id',
    accountId: 'mock-account-id',
    roleArn: 'mock-role-arn',
    pipelineArtifacts: {
      'remoter-server': 'mock-remoter-server-image',
      'remoter-client': 'mock-remoter-client-image',
    },
    clusterInfo: {
      name: 'mock-cluster-name',
      endpoint: 'mock-endpoint',
      certAuthority: 'mock-ca',
    },
    processingConfig: {},
  };

  it('- qc local development', () => {
    const qcPipelineSkeleton = getQcPipelineSkeleton('development');
    const stateMachine = buildStateMachineDefinition(qcPipelineSkeleton, context);
    expect(stateMachine).toMatchSnapshot();
  });

  it('- qc staging', () => {
    const qcPipelineSkeleton = getQcPipelineSkeleton('staging');
    const stateMachine = buildStateMachineDefinition(qcPipelineSkeleton, context);
    expect(stateMachine).toMatchSnapshot();
  });

  it('- qc production', () => {
    const qcPipelineSkeleton = getQcPipelineSkeleton('production');
    const stateMachine = buildStateMachineDefinition(qcPipelineSkeleton, context);
    expect(stateMachine).toMatchSnapshot();
  });

  it('- gem2s local development', () => {
    const gem2sPipelineSkeleton = getGem2sPipelineSkeleton('development');
    const stateMachine = buildStateMachineDefinition(gem2sPipelineSkeleton, context);
    expect(stateMachine).toMatchSnapshot();
  });

  it('- gem2s staging', () => {
    const gem2sPipelineSkeleton = getGem2sPipelineSkeleton('staging');
    const stateMachine = buildStateMachineDefinition(gem2sPipelineSkeleton, context);
    expect(stateMachine).toMatchSnapshot();
  });

  it('- gem2s production', () => {
    const gem2sPipelineSkeleton = getGem2sPipelineSkeleton('production');
    const stateMachine = buildStateMachineDefinition(gem2sPipelineSkeleton, context);
    expect(stateMachine).toMatchSnapshot();
  });
});
