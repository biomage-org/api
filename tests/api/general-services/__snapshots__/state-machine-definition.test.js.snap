// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`non-tests to document the State Machines - gem2s cloud 1`] = `

{
  "Comment": "Gem2s for clusterEnv 'test'",
  "StartAt": "DeleteCompletedGem2SWorker",
  "States": {
    "DeleteCompletedGem2SWorker": {
      "Next": "LaunchNewGem2SWorker",
      "ResultPath": null,
      "Type": "Task",
      "Comment": "Deletes the previous server pipeline HelmRelease (Service+Job).",
      "Resource": "arn:aws:states:::eks:call",
      "Parameters": {
        "ClusterName": "mock-cluster-name",
        "CertificateAuthority": "mock-ca",
        "Endpoint": "mock-endpoint",
        "Method": "DELETE",
        "Path": "/apis/helm.fluxcd.io/v1/namespaces/pipeline-test-namespace/helmreleases",
        "QueryParameters": {
          "labelSelector": [
            "type=pipeline,sandboxId=default,experimentId=mock-experiment-id"
          ]
        }
      }
    },
    "LaunchNewGem2SWorker": {
      "Next": "DownloadGem",
      "ResultPath": null,
      "Type": "Task",
      "Comment": "Attempts to create a Kubernetes Job+Service for the pipeline runner. Will swallow a 409 (already exists) error.",
      "Resource": "arn:aws:states:::eks:call",
      "Parameters": {
        "ClusterName": "mock-cluster-name",
        "CertificateAuthority": "mock-ca",
        "Endpoint": "mock-endpoint",
        "Method": "POST",
        "Path": "/apis/helm.fluxcd.io/v1/namespaces/pipeline-test-namespace/helmreleases",
        "RequestBody": {
          "apiVersion": "helm.fluxcd.io/v1",
          "kind": "HelmRelease",
          "metadata": {
            "name": "undefined-mock-experiment-id",
            "namespace": "pipeline-test-namespace",
            "annotations": {
              "fluxcd.io/automated": "true"
            },
            "labels": {
              "sandboxId": "default",
              "type": "pipeline",
              "experimentId": "mock-experiment-id"
            }
          },
          "spec": {
            "releaseName": "undefined-mock-experiment-id",
            "chart": {
              "git": "git@github.com:biomage-ltd/pipeline",
              "path": "pipeline-runner/chart"
            },
            "values": {
              "experimentId": "mock-experiment-id",
              "namespace": "pipeline-test-namespace",
              "sandboxId": "default",
              "awsAccountId": "mock-account-id",
              "clusterEnv": "test",
              "awsRegion": "eu-west-1"
            }
          }
        }
      },
      "Retry": [
        {
          "ErrorEquals": [
            "EKS.409"
          ],
          "IntervalSeconds": 1,
          "BackoffRate": 2,
          "MaxAttempts": 2
        }
      ],
      "Catch": [
        {
          "ErrorEquals": [
            "EKS.409"
          ],
          "ResultPath": "$.error-info",
          "Next": "DownloadGem"
        }
      ]
    },
    "DownloadGem": {
      "Next": "PreProcessing",
      "Type": "Task",
      "ResultPath": null,
      "TimeoutSeconds": 10800,
      "Parameters": {
        "experimentId": "mock-experiment-id",
        "taskName": "downloadGem",
        "config": {},
        "server": "remoter-server-mock-experiment-id.pipeline-test-namespace.svc.cluster.local"
      }
    },
    "PreProcessing": {
      "Next": "EmptyDrops",
      "Type": "Task",
      "ResultPath": null,
      "TimeoutSeconds": 10800,
      "Parameters": {
        "experimentId": "mock-experiment-id",
        "taskName": "preproc",
        "config": {},
        "server": "remoter-server-mock-experiment-id.pipeline-test-namespace.svc.cluster.local"
      }
    },
    "EmptyDrops": {
      "Next": "DoubletScores",
      "Type": "Task",
      "ResultPath": null,
      "TimeoutSeconds": 10800,
      "Parameters": {
        "experimentId": "mock-experiment-id",
        "taskName": "emptyDrops",
        "config": {},
        "server": "remoter-server-mock-experiment-id.pipeline-test-namespace.svc.cluster.local"
      }
    },
    "DoubletScores": {
      "Next": "CreateSeurat",
      "Type": "Task",
      "ResultPath": null,
      "TimeoutSeconds": 10800,
      "Parameters": {
        "experimentId": "mock-experiment-id",
        "taskName": "doubletScores",
        "config": {},
        "server": "remoter-server-mock-experiment-id.pipeline-test-namespace.svc.cluster.local"
      }
    },
    "CreateSeurat": {
      "Next": "PrepareExperiment",
      "Type": "Task",
      "ResultPath": null,
      "TimeoutSeconds": 10800,
      "Parameters": {
        "experimentId": "mock-experiment-id",
        "taskName": "createSeurat",
        "config": {},
        "server": "remoter-server-mock-experiment-id.pipeline-test-namespace.svc.cluster.local"
      }
    },
    "PrepareExperiment": {
      "Next": "UploadToAWS",
      "Type": "Task",
      "ResultPath": null,
      "TimeoutSeconds": 10800,
      "Parameters": {
        "experimentId": "mock-experiment-id",
        "taskName": "prepareExperiment",
        "config": {},
        "server": "remoter-server-mock-experiment-id.pipeline-test-namespace.svc.cluster.local"
      }
    },
    "UploadToAWS": {
      "Next": "EndOfGem2S",
      "Type": "Task",
      "ResultPath": null,
      "TimeoutSeconds": 10800,
      "Parameters": {
        "experimentId": "mock-experiment-id",
        "taskName": "uploadToAWS",
        "config": {},
        "server": "remoter-server-mock-experiment-id.pipeline-test-namespace.svc.cluster.local"
      }
    },
    "EndOfGem2S": {
      "Type": "Pass",
      "End": true
    }
  }
}
`;

exports[`non-tests to document the State Machines - gem2s local development 1`] = `

{
  "Comment": "Gem2s for clusterEnv 'test'",
  "StartAt": "DeleteCompletedGem2SWorker",
  "States": {
    "DeleteCompletedGem2SWorker": {
      "Next": "LaunchNewGem2SWorker",
      "ResultPath": null,
      "Type": "Task",
      "Comment": "Removes Docker containers with pipeline runs on the local machine.",
      "Resource": "arn:aws:states:::lambda:invoke",
      "Parameters": {
        "FunctionName": "arn:aws:lambda:eu-west-1:mock-account-id:function:remove-previous-pipeline-containers"
      }
    },
    "LaunchNewGem2SWorker": {
      "Next": "DownloadGem",
      "ResultPath": null,
      "Type": "Task",
      "Resource": "arn:aws:states:::lambda:invoke",
      "Parameters": {
        "FunctionName": "arn:aws:lambda:eu-west-1:mock-account-id:function:local-container-launcher",
        "Payload": {
          "image": "biomage-pipeline-runner",
          "name": "undefined-runner",
          "detached": true
        }
      },
      "Catch": [
        {
          "ErrorEquals": [
            "States.ALL"
          ],
          "ResultPath": "$.error-info",
          "Next": "DownloadGem"
        }
      ]
    },
    "DownloadGem": {
      "Next": "PreProcessing",
      "Type": "Task",
      "ResultPath": null,
      "TimeoutSeconds": 10800,
      "Parameters": {
        "experimentId": "mock-experiment-id",
        "taskName": "downloadGem",
        "config": {},
        "server": "host.docker.internal"
      }
    },
    "PreProcessing": {
      "Next": "EmptyDrops",
      "Type": "Task",
      "ResultPath": null,
      "TimeoutSeconds": 10800,
      "Parameters": {
        "experimentId": "mock-experiment-id",
        "taskName": "preproc",
        "config": {},
        "server": "host.docker.internal"
      }
    },
    "EmptyDrops": {
      "Next": "DoubletScores",
      "Type": "Task",
      "ResultPath": null,
      "TimeoutSeconds": 10800,
      "Parameters": {
        "experimentId": "mock-experiment-id",
        "taskName": "emptyDrops",
        "config": {},
        "server": "host.docker.internal"
      }
    },
    "DoubletScores": {
      "Next": "CreateSeurat",
      "Type": "Task",
      "ResultPath": null,
      "TimeoutSeconds": 10800,
      "Parameters": {
        "experimentId": "mock-experiment-id",
        "taskName": "doubletScores",
        "config": {},
        "server": "host.docker.internal"
      }
    },
    "CreateSeurat": {
      "Next": "PrepareExperiment",
      "Type": "Task",
      "ResultPath": null,
      "TimeoutSeconds": 10800,
      "Parameters": {
        "experimentId": "mock-experiment-id",
        "taskName": "createSeurat",
        "config": {},
        "server": "host.docker.internal"
      }
    },
    "PrepareExperiment": {
      "Next": "UploadToAWS",
      "Type": "Task",
      "ResultPath": null,
      "TimeoutSeconds": 10800,
      "Parameters": {
        "experimentId": "mock-experiment-id",
        "taskName": "prepareExperiment",
        "config": {},
        "server": "host.docker.internal"
      }
    },
    "UploadToAWS": {
      "Next": "EndOfGem2S",
      "Type": "Task",
      "ResultPath": null,
      "TimeoutSeconds": 10800,
      "Parameters": {
        "experimentId": "mock-experiment-id",
        "taskName": "uploadToAWS",
        "config": {},
        "server": "host.docker.internal"
      }
    },
    "EndOfGem2S": {
      "Type": "Pass",
      "End": true
    }
  }
}
`;

exports[`non-tests to document the State Machines - qc cloud 1`] = `

{
  "Comment": "Pipeline for clusterEnv 'test'",
  "StartAt": "DeleteCompletedPipelineWorker",
  "States": {
    "DeleteCompletedPipelineWorker": {
      "Next": "LaunchNewPipelineWorker",
      "ResultPath": null,
      "Type": "Task",
      "Comment": "Deletes the previous server pipeline HelmRelease (Service+Job).",
      "Resource": "arn:aws:states:::eks:call",
      "Parameters": {
        "ClusterName": "mock-cluster-name",
        "CertificateAuthority": "mock-ca",
        "Endpoint": "mock-endpoint",
        "Method": "DELETE",
        "Path": "/apis/helm.fluxcd.io/v1/namespaces/pipeline-test-namespace/helmreleases",
        "QueryParameters": {
          "labelSelector": [
            "type=pipeline,sandboxId=default,experimentId=mock-experiment-id"
          ]
        }
      }
    },
    "LaunchNewPipelineWorker": {
      "Next": "ClassifierFilterMap",
      "ResultPath": null,
      "Type": "Task",
      "Comment": "Attempts to create a Kubernetes Job+Service for the pipeline runner. Will swallow a 409 (already exists) error.",
      "Resource": "arn:aws:states:::eks:call",
      "Parameters": {
        "ClusterName": "mock-cluster-name",
        "CertificateAuthority": "mock-ca",
        "Endpoint": "mock-endpoint",
        "Method": "POST",
        "Path": "/apis/helm.fluxcd.io/v1/namespaces/pipeline-test-namespace/helmreleases",
        "RequestBody": {
          "apiVersion": "helm.fluxcd.io/v1",
          "kind": "HelmRelease",
          "metadata": {
            "name": "undefined-mock-experiment-id",
            "namespace": "pipeline-test-namespace",
            "annotations": {
              "fluxcd.io/automated": "true"
            },
            "labels": {
              "sandboxId": "default",
              "type": "pipeline",
              "experimentId": "mock-experiment-id"
            }
          },
          "spec": {
            "releaseName": "undefined-mock-experiment-id",
            "chart": {
              "git": "git@github.com:biomage-ltd/pipeline",
              "path": "pipeline-runner/chart"
            },
            "values": {
              "experimentId": "mock-experiment-id",
              "namespace": "pipeline-test-namespace",
              "sandboxId": "default",
              "awsAccountId": "mock-account-id",
              "clusterEnv": "test",
              "awsRegion": "eu-west-1"
            }
          }
        }
      },
      "Retry": [
        {
          "ErrorEquals": [
            "EKS.409"
          ],
          "IntervalSeconds": 1,
          "BackoffRate": 2,
          "MaxAttempts": 2
        }
      ],
      "Catch": [
        {
          "ErrorEquals": [
            "EKS.409"
          ],
          "ResultPath": "$.error-info",
          "Next": "ClassifierFilterMap"
        }
      ]
    },
    "ClassifierFilterMap": {
      "Type": "Map",
      "Next": "CellSizeDistributionFilterMap",
      "ResultPath": null,
      "ItemsPath": "$.samples",
      "Iterator": {
        "StartAt": "ClassifierFilter",
        "States": {
          "ClassifierFilter": {
            "End": true,
            "Type": "Task",
            "ResultPath": null,
            "TimeoutSeconds": 10800,
            "Parameters": {
              "experimentId": "mock-experiment-id",
              "taskName": "classifier",
              "config": {},
              "server": "remoter-server-mock-experiment-id.pipeline-test-namespace.svc.cluster.local"
            }
          }
        }
      }
    },
    "CellSizeDistributionFilterMap": {
      "Type": "Map",
      "Next": "MitochondrialContentFilterMap",
      "ResultPath": null,
      "ItemsPath": "$.samples",
      "Iterator": {
        "StartAt": "CellSizeDistributionFilter",
        "States": {
          "CellSizeDistributionFilter": {
            "End": true,
            "Type": "Task",
            "ResultPath": null,
            "TimeoutSeconds": 10800,
            "Parameters": {
              "experimentId": "mock-experiment-id",
              "taskName": "cellSizeDistribution",
              "config": {},
              "server": "remoter-server-mock-experiment-id.pipeline-test-namespace.svc.cluster.local"
            }
          }
        }
      }
    },
    "MitochondrialContentFilterMap": {
      "Type": "Map",
      "Next": "NumGenesVsNumUmisFilterMap",
      "ResultPath": null,
      "ItemsPath": "$.samples",
      "Iterator": {
        "StartAt": "MitochondrialContentFilter",
        "States": {
          "MitochondrialContentFilter": {
            "End": true,
            "Type": "Task",
            "ResultPath": null,
            "TimeoutSeconds": 10800,
            "Parameters": {
              "experimentId": "mock-experiment-id",
              "taskName": "mitochondrialContent",
              "config": {},
              "server": "remoter-server-mock-experiment-id.pipeline-test-namespace.svc.cluster.local"
            }
          }
        }
      }
    },
    "NumGenesVsNumUmisFilterMap": {
      "Type": "Map",
      "Next": "DoubletScoresFilterMap",
      "ResultPath": null,
      "ItemsPath": "$.samples",
      "Iterator": {
        "StartAt": "NumGenesVsNumUmisFilter",
        "States": {
          "NumGenesVsNumUmisFilter": {
            "End": true,
            "Type": "Task",
            "ResultPath": null,
            "TimeoutSeconds": 10800,
            "Parameters": {
              "experimentId": "mock-experiment-id",
              "taskName": "numGenesVsNumUmis",
              "config": {},
              "server": "remoter-server-mock-experiment-id.pipeline-test-namespace.svc.cluster.local"
            }
          }
        }
      }
    },
    "DoubletScoresFilterMap": {
      "Type": "Map",
      "Next": "DataIntegration",
      "ResultPath": null,
      "ItemsPath": "$.samples",
      "Iterator": {
        "StartAt": "DoubletScoresFilter",
        "States": {
          "DoubletScoresFilter": {
            "End": true,
            "Type": "Task",
            "ResultPath": null,
            "TimeoutSeconds": 10800,
            "Parameters": {
              "experimentId": "mock-experiment-id",
              "taskName": "doubletScores",
              "config": {},
              "server": "remoter-server-mock-experiment-id.pipeline-test-namespace.svc.cluster.local"
            }
          }
        }
      }
    },
    "DataIntegration": {
      "Next": "ConfigureEmbedding",
      "Type": "Task",
      "ResultPath": null,
      "TimeoutSeconds": 10800,
      "Parameters": {
        "experimentId": "mock-experiment-id",
        "taskName": "dataIntegration",
        "config": {},
        "server": "remoter-server-mock-experiment-id.pipeline-test-namespace.svc.cluster.local"
      }
    },
    "ConfigureEmbedding": {
      "Next": "EndOfPipeline",
      "Type": "Task",
      "ResultPath": null,
      "TimeoutSeconds": 10800,
      "Parameters": {
        "experimentId": "mock-experiment-id",
        "taskName": "configureEmbedding",
        "config": {},
        "server": "remoter-server-mock-experiment-id.pipeline-test-namespace.svc.cluster.local"
      }
    },
    "EndOfPipeline": {
      "Type": "Pass",
      "End": true
    }
  }
}
`;

exports[`non-tests to document the State Machines - qc local development 1`] = `

{
  "Comment": "Pipeline for clusterEnv 'test'",
  "StartAt": "DeleteCompletedPipelineWorker",
  "States": {
    "DeleteCompletedPipelineWorker": {
      "Next": "LaunchNewPipelineWorker",
      "ResultPath": null,
      "Type": "Task",
      "Comment": "Removes Docker containers with pipeline runs on the local machine.",
      "Resource": "arn:aws:states:::lambda:invoke",
      "Parameters": {
        "FunctionName": "arn:aws:lambda:eu-west-1:mock-account-id:function:remove-previous-pipeline-containers"
      }
    },
    "LaunchNewPipelineWorker": {
      "Next": "ClassifierFilterMap",
      "ResultPath": null,
      "Type": "Task",
      "Resource": "arn:aws:states:::lambda:invoke",
      "Parameters": {
        "FunctionName": "arn:aws:lambda:eu-west-1:mock-account-id:function:local-container-launcher",
        "Payload": {
          "image": "biomage-pipeline-runner",
          "name": "undefined-runner",
          "detached": true
        }
      },
      "Catch": [
        {
          "ErrorEquals": [
            "States.ALL"
          ],
          "ResultPath": "$.error-info",
          "Next": "ClassifierFilterMap"
        }
      ]
    },
    "ClassifierFilterMap": {
      "Type": "Map",
      "Next": "CellSizeDistributionFilterMap",
      "ResultPath": null,
      "ItemsPath": "$.samples",
      "Iterator": {
        "StartAt": "ClassifierFilter",
        "States": {
          "ClassifierFilter": {
            "End": true,
            "Type": "Task",
            "ResultPath": null,
            "TimeoutSeconds": 10800,
            "Parameters": {
              "experimentId": "mock-experiment-id",
              "taskName": "classifier",
              "config": {},
              "server": "host.docker.internal"
            }
          }
        }
      }
    },
    "CellSizeDistributionFilterMap": {
      "Type": "Map",
      "Next": "MitochondrialContentFilterMap",
      "ResultPath": null,
      "ItemsPath": "$.samples",
      "Iterator": {
        "StartAt": "CellSizeDistributionFilter",
        "States": {
          "CellSizeDistributionFilter": {
            "End": true,
            "Type": "Task",
            "ResultPath": null,
            "TimeoutSeconds": 10800,
            "Parameters": {
              "experimentId": "mock-experiment-id",
              "taskName": "cellSizeDistribution",
              "config": {},
              "server": "host.docker.internal"
            }
          }
        }
      }
    },
    "MitochondrialContentFilterMap": {
      "Type": "Map",
      "Next": "NumGenesVsNumUmisFilterMap",
      "ResultPath": null,
      "ItemsPath": "$.samples",
      "Iterator": {
        "StartAt": "MitochondrialContentFilter",
        "States": {
          "MitochondrialContentFilter": {
            "End": true,
            "Type": "Task",
            "ResultPath": null,
            "TimeoutSeconds": 10800,
            "Parameters": {
              "experimentId": "mock-experiment-id",
              "taskName": "mitochondrialContent",
              "config": {},
              "server": "host.docker.internal"
            }
          }
        }
      }
    },
    "NumGenesVsNumUmisFilterMap": {
      "Type": "Map",
      "Next": "DoubletScoresFilterMap",
      "ResultPath": null,
      "ItemsPath": "$.samples",
      "Iterator": {
        "StartAt": "NumGenesVsNumUmisFilter",
        "States": {
          "NumGenesVsNumUmisFilter": {
            "End": true,
            "Type": "Task",
            "ResultPath": null,
            "TimeoutSeconds": 10800,
            "Parameters": {
              "experimentId": "mock-experiment-id",
              "taskName": "numGenesVsNumUmis",
              "config": {},
              "server": "host.docker.internal"
            }
          }
        }
      }
    },
    "DoubletScoresFilterMap": {
      "Type": "Map",
      "Next": "DataIntegration",
      "ResultPath": null,
      "ItemsPath": "$.samples",
      "Iterator": {
        "StartAt": "DoubletScoresFilter",
        "States": {
          "DoubletScoresFilter": {
            "End": true,
            "Type": "Task",
            "ResultPath": null,
            "TimeoutSeconds": 10800,
            "Parameters": {
              "experimentId": "mock-experiment-id",
              "taskName": "doubletScores",
              "config": {},
              "server": "host.docker.internal"
            }
          }
        }
      }
    },
    "DataIntegration": {
      "Next": "ConfigureEmbedding",
      "Type": "Task",
      "ResultPath": null,
      "TimeoutSeconds": 10800,
      "Parameters": {
        "experimentId": "mock-experiment-id",
        "taskName": "dataIntegration",
        "config": {},
        "server": "host.docker.internal"
      }
    },
    "ConfigureEmbedding": {
      "Next": "EndOfPipeline",
      "Type": "Task",
      "ResultPath": null,
      "TimeoutSeconds": 10800,
      "Parameters": {
        "experimentId": "mock-experiment-id",
        "taskName": "configureEmbedding",
        "config": {},
        "server": "host.docker.internal"
      }
    },
    "EndOfPipeline": {
      "Type": "Pass",
      "End": true
    }
  }
}
`;
